### PR TITLE
Added Security Rules for Firestore

### LINKED ISSUE #21: Implement Firebase Security Rules and Add Tests
We should have security rules in place to avoid unauthorized access to our data by unauthenticated/disabled users and to also enforce some generally good practices.

## Requirements
- only allow authenticated and enabled users to access any data
- user profile data is only written by Cloud Functions, so all user data (including decks for the time being), should be read only (Cloud Functions use the server sdk which bypass these rules)
- only allow authenticated users in a game to write any data (outside users can read, e.g. spectate)
  - they can create cards (e.g. tokens)
  - they can change their own life totals and counters (player data)
- only allow a card's _controller_ to write to it (e.g. only I can tap my lands, no one else)

### PR DESCRIPTION
**Pull request by eeeeeric**
Added security rules for Firestore such that:
- All reads require authentication (writes require authentiation and more)
- Updates to a player's own data for a game can only be made by that player
- Only players in a game can create additional cards in the game, and those cards must go to the Battlefield (the public zone). Note that currently, reads are allowed for the entire game, meaning a player does have read access to another player's cards in hand.
- Only a card's controller may update it (e.g. tap it)

Resolves #21

### CODE REVIEW

File: firebase/firestore.rules [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
+
+      match /Players/{player} {
+        // Only players may update their own data
+        allow update: if request.auth.uid == player;
+      }
+
+      match /Cards/{card} {
+        // Only players in the game may create new cards (e.g. tokens)
+        allow create: if get(/databases/$(database)/documents/Games/$(game)/Players/$(request.auth.uid)) != null && 
+          request.resource.data.state.zone == 'Battlefield';
```
Comment (hrecker) [inline firebase/firestore.rules]: I assume this doesn't cause any problem with the cloud functions creating new cards?

**Comment by hrecker** [comment]
I think allowing read access to all cards including other players hands is good. I think it's more on the UI to keep that from happening unintentionally, personally I'm not really concerned with making this app hard to hack, since I'm assuming people won't use it to cheat.

File: firebase/firestore.rules (hunk 1 above)
Comment (eeeeeric) [reply firebase/firestore.rules]: The Cloud Functions aren't subject to security rules so they can continue to create cards.

**Comment by eeeeeric** [comment]
Yeah, I don't think it is a problem we need to solve for our use case. Thanks for taking the time to review.

**Pull request merged**